src/runtime/components/Table.vue

```diff
@@ -88,8 +90,10 @@ export interface TableProps<T> {
 }
 
 export interface TableSlots<T> {
-  expanded(props: { row: Row<T> }): any
-  empty(props?: {}): any
+  [key: `${string}-header`]: (props: HeaderContext<T, unknown>) => any
```

Comment:
`DynamicSlots` helper could not be reused because it is a composite key made of a literal and a props key.

I did not make a universal type, readability drops very much. I left 2 different types for header and cell.